src/ci/docker/host-x86_64/x86_64-gnu-llvm-18/Dockerfile

```diff
@@ -51,6 +51,7 @@ ENV RUST_CONFIGURE_ARGS \
       --build=x86_64-unknown-linux-gnu \
       --llvm-root=/usr/lib/llvm-18 \
       --enable-llvm-link-shared \
+      --set rust.lld=false \
```

Comment:
Hmm, if an external LLVM is used, it shouldn't even be possible to build LLD, IMO. This goes back to the discussion that perhaps the `lld` config should be under `llvm` and it should just mean "build LLD" and nothing more :) Anyway, that's just me thinking out loud.

Reply:
> This goes back to the discussion that perhaps the lld config should be under llvm and it should just mean "build LLD" and nothing more :)

I think the same way, but the scope of this PR is kind a different (utilizing lld_enabled flag for dist tarball).

Reply:
What do you think about https://github.com/rust-lang/rust/pull/126701/commits/7aec1b01bbb9d06956819f34371cf071858d2f31 ?

Reply:
Oops, I completely missed this comment, sorry. Yeah, that makes sense. If we use an external LLVM, we shouldn't build LLD (that's also why LLD should be in the `llvm` section, IMO :) ).